### DISCUSSION TITLE
Ho to use aws_completer

### DISCUSSION THREAD

**Discussion by aviv-akhansari** [nushell/nushell#13038] [Q&A]
Hi,
Is there any way to use [aws_completer](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-completion.html) with nushell?

Edit:

I ended up doing this:
```nu
def "nu-complete aws" [input: string] {
  with-env { COMMAND_LINE: $input } { aws_completer | lines }
}

export extern "aws" [
  command?: string@"nu-complete aws"
]
```

But it works only for the first command/input.
How can I do this recursively for all nested commands and parameters?

**Comment by aviv-akhansari** [comment]
I finally use https://carapace.sh instead and also removed some custom completions I got from https://github.com/nushell/nu_scripts.
I'll see in the long term if it's a good solution or not but it seems very promising.
https://carapace-sh.github.io/carapace-bin/setup.html#nushell

**Comment by Esgariot** [comment]
```diff
export extern "aws" [
-  command?: string@"nu-complete aws"
+  ...command: string@"nu-complete aws"
]
```

It completes subcommands, e.g. `aws s3 <TAB>` lists `ls`, `cp` ...
